project/project_test.go

```diff
@@ -19,3 +20,27 @@ func TestEventEquality(t *testing.T) {
 		t.Fatal("Events match")
 	}
 }
+
+func TestBadFile(t *testing.T) {
+	p := NewProject(&Context{
+		ComposeBytes: []byte("garbage"),
+	})
+
+	err := p.Parse()
+	if err == nil {
+		t.Fatal("Should have failed parse")
+	}
+
+	if !strings.HasPrefix(err.Error(), "yaml: unmarshal errors") {
+		t.Fatal("Should have failed parse", err)
+	}
+
+	p = NewProject(&Context{
+		ComposeBytes: []byte("not-garbage:\n  image: foo"),
+	})
+
+	err = p.Parse()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
I always put the "success" case and "failure" case in separate tests so the output summary is more clear.  I'm not sure if this is a common practice in golang or not.